### PR TITLE
feat: Optimized data loading for PyTorch

### LINKED ISSUE #190: [datasets] Optimize dataloaders for faster iterations
The current training scripts in `references` are way too slow.
Let's discuss here how we can improve them

### PR DESCRIPTION
**Pull request by fg-mindee**
Following up on #190, this PR introduces the following modifications:
- added automatic workers number selection for PyTorch trainings
- switched image reading backend from torchvision to PIL + tensor conversion
- reflected changes on training scripts
- had to update PIL requirements due to https://github.com/python-pillow/Pillow/issues/5571

Iterating with the dataloader on FUNSD gets a 25%+ speedup with the same number of workers (3X compared to the original default number of workers)

Closes #190

Any feedback is welcome!

### CODE REVIEW

File: references/detection/train_pytorch.py [hunk 1]
```diff
@@ -77,16 +78,16 @@ def main(args):
...
+        args.workers = min(16, mp.cpu_count())
+
     torch.backends.cudnn.benchmark = True
 
     st = time.time()
     val_set = DetectionDataset(
         img_folder=os.path.join(args.data_path, 'val'),
         label_folder=os.path.join(args.data_path, 'val_labels'),
-        sample_transforms=Compose([
-            Lambda(lambda x: x / 255),
```
Comment (charlesmindee) [inline references/detection/train_pytorch.py:87]: Why do you remove the normalization here ?

File: references/detection/train_pytorch.py [hunk 2]
```diff
@@ -137,7 +138,6 @@ def main(args):
         img_folder=os.path.join(args.data_path, 'train'),
         label_folder=os.path.join(args.data_path, 'train_labels'),
         sample_transforms=Compose([
-            Lambda(lambda x: x / 255),
```
Comment (charlesmindee) [inline references/detection/train_pytorch.py:140]: same here

File: references/detection/train_pytorch.py (hunk 1 above)
Comment (fg-mindee) [reply references/detection/train_pytorch.py:87]: The  `to_tensor` in the dataset `__getitem__` already does that :)

File: references/detection/train_pytorch.py (hunk 2 above)
Comment (fg-mindee) [reply references/detection/train_pytorch.py:140]: cf. previous comment

**Review by charlesmindee — APPROVED** [review]
